src/libraries/System.IO.IsolatedStorage/src/System/IO/IsolatedStorage/Helper.Win32Unix.cs

```diff
@@ -51,21 +51,30 @@ internal static void GetDefaultIdentityAndHash(out object identity, out string h
             Assembly? assembly = Assembly.GetEntryAssembly();
 
             if (assembly == null)
+            {
                 throw new IsolatedStorageException(SR.IsolatedStorage_Init);
+            }
 
             AssemblyName assemblyName = assembly.GetName();
-#pragma warning disable SYSLIB0012
-            Uri codeBase = new Uri(assembly.CodeBase!);
-#pragma warning restore SYSLIB0012
-
             hash = IdentityHelper.GetNormalizedStrongNameHash(assemblyName)!;
+
             if (hash != null)
             {
                 hash = "StrongName" + separator + hash;
                 identity = assemblyName;
             }
+            // in case of SingleFile, Location.Length is 0, we will use native entrypoint's filename.
+            else if (assembly.Location.Length == 0)
+            {
+                var assemblyPath = new Uri(Path.Combine(AppContext.BaseDirectory, assemblyName.Name!) + ExeFileExtension);
+                hash = "Url" + separator + IdentityHelper.GetNormalizedUriHash(assemblyPath);
+                identity = assemblyPath;
+            }
             else
             {
+#pragma warning disable SYSLIB0012
+                Uri codeBase = new Uri(assembly.CodeBase!);
```

Comment:
Will this only ever throw on single file?  If not it could make sense to just use new approach all the time?

Reply:
With the matrix we are currently testing, single file (or more specifically in-memory assembly) is the only configuration which is throwing from CodeBase property since https://github.com/dotnet/runtime/commit/0546bd55c24618846d487d1315cf1d8f47dad4de. Perhaps using the new approach regardless, and avoiding the deprecated API usage would still make sense? One downside with new approach is the case where someone has renamed `<name>.exe` with different extension, and we will end up using the standard `.exe` extension on Windows.

Reply:
Actually this normally uses `.dll` for .NETCore applications since the app assembly itself is `.dll`. 

The purpose of this API is to provide seed-data to a hashing function to produce a stable hash.  So long as it is stable when inputs don't change it should be fine, we could actually omit the extension since it is hardcoded and never changes.  

Ideally we should not change the location that the hash produces if the inputs do not change.  This ensures an application which only updates System.IO.IsolatedStorage does not lose access to its files.  If we could come up with an algorithm that produced the same result as assembly.Codebase for the existing apps and then just something that was stable for the single-file app that would be sufficient.

Reply:
Thanks. In case of true SingleFile, the file extension would be .exe on Windows and empty string on Unix. For non-single-file case, it is .dll for both Windows and Unix. I have changed the algorithm accordingly and removed the and removed the branch which uses `CodeBase`. We can also just use .dll extension in all cases, agnostic of OS, since it is only used as seed-data as you have described. Would that be a cleaner approach?

Reply:
Almost, I was bringing up the `.dll` case just to point out a difference.  I think your point about rename is legitimate.  I have a small suggestion which I think might strike the right balance.

Reply:
I see your point about omitting the extension entirely. That has made the change simpler. :)